### DISCUSSION TITLE
How nextjs layout entity works?

### DISCUSSION THREAD

**Discussion by lior-amsalem** [vercel/next.js#70907] [Help]
### Summary

For whatever the reason i thought that the layout is wrapping the page entity out of the box? and it's children.

is that not the case?

let me explain:

/app/layout.tsx <-- main layout
/app/page.tsx <-- will have above layout.
/app/my-content/page.tsx
/app/my-content/layout.tsx <-- sub layout, it will wrap current page tsx and forward it onboard to the next layout entity
/app/my-content/my-sub-content/page.tsx
/app/my-content/my-sub-content/layout.tsx <--- will warp above page and forward it throught second and first layouts as well as a wrappers.

thats not the case ? imitating the nature of html and dom

**Comment by icyJoseph** [comment]
Show us some code.

> Layout components should accept and use a children prop. During rendering, children will be populated with the route segments the layout is wrapping. These will primarily be the component of a child [Layout](https://nextjs.org/docs/app/building-your-application/routing/pages-and-layouts#pages) (if it exists) or [Page](https://nextjs.org/docs/app/building-your-application/routing/pages-and-layouts#pages), but could also be other special files like [Loading](https://nextjs.org/docs/app/building-your-application/routing/loading-ui-and-streaming) or [Error](https://nextjs.org/docs/app/building-your-application/routing/error-handling) when applicable.
